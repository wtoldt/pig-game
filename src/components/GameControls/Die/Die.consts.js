export const DIE_VALUES = {
  1: [
    ' ',    ' ',    ' ',
    ' ',    '🐽',   ' ',
    ' ',    ' ',    ' '
  ], 2: [
    '🔴',    ' ',    ' ',
    ' ',    ' ',    ' ',
    ' ',    ' ',    '🔴'
  ], 3: [
    '🔴',    ' ',    ' ',
    ' ',    '🔴',    ' ',
    ' ',    ' ',    '🔴'
  ], 4: [
    '🔴',    ' ',    '🔴',
    ' ',    ' ',    ' ',
    '🔴',    ' ',    '🔴'
  ], 5: [
    '🔴',    ' ',    '🔴',
    ' ',    '🔴',    ' ',
    '🔴',    ' ',    '🔴'
  ], 6: [
    '🔴',    ' ',    '🔴',
    '🔴',    ' ',    '🔴',
    '🔴',    ' ',    '🔴'
  ],
};