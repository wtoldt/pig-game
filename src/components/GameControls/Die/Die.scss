@import '../../../styles/consts.scss';

.pg-die {
  height: 50px;
  width: 50px;
  background-color: ivory;
  border: 3px solid black;
  border-radius: $big-panel-br;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  .row {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    align-items: center;

    .cell {
      flex: 1 1;
      color: red;
      font-family: monospace;
      height: 17px;
      width: 17px
    }
  }
}