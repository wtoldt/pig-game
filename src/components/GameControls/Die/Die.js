import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { DIE_VALUES } from './Die.consts';

import './Die.scss';

export default class Die extends Component {

  static propTypes = {
    value: PropTypes.number.isRequired
  };

  static defaultProps = {
    value: 1
  };

  render() {
    const { value } = this.props;
    const [c1, c2, c3, c4, c5, c6, c7, c8, c9] = DIE_VALUES[value];
    const classnames = classNames('pg-die');
    return (
      <div className={classnames}>
        <div className="row">
          <div id="c1" className="cell">{c1}</div>
          <div id="c2" className="cell">{c2}</div>
          <div id="c3" className="cell">{c3}</div>
        </div>
        <div className="row">
          <div id="c4" className="cell">{c4}</div>
          <div id="c5" className="cell">{c5}</div>
          <div id="c6" className="cell">{c6}</div>
        </div>
        <div className="row">
          <div id="c7" className="cell">{c7}</div>
          <div id="c8" className="cell">{c8}</div>
          <div id="c9" className="cell">{c9}</div>
        </div>
      </div>
    );
  }
}