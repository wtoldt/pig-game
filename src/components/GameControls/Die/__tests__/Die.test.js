import { render } from '@testing-library/react';
import Die from '../Die';

const defaultProps = {
  value: 1
};

function getWrapper(props={}) {
  return render(
    <Die {...defaultProps} {...props} />
  );
}

describe('Die Component', () => {

  describe('Snapshot testing', () => {
    it('Default snapshot', () => {
      const { container } = getWrapper();
      expect(container).toMatchSnapshot();
    });

    //test 1-6
    it.each([1, 2, 3, 4, 5, 6])('%i snapshot test', (value) => {
      const { container } = getWrapper({ value });
      expect(container).toMatchSnapshot();
    });

  });
});
