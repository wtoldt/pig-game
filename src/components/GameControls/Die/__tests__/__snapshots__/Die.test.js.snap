// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Die Component Snapshot testing 1 snapshot test 1`] = `
<div>
  <div
    class="pg-die"
  >
    <div
      class="row"
    >
      <div
        class="cell"
        id="c1"
      >
         
      </div>
      <div
        class="cell"
        id="c2"
      >
         
      </div>
      <div
        class="cell"
        id="c3"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c4"
      >
         
      </div>
      <div
        class="cell"
        id="c5"
      >
        🐽
      </div>
      <div
        class="cell"
        id="c6"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c7"
      >
         
      </div>
      <div
        class="cell"
        id="c8"
      >
         
      </div>
      <div
        class="cell"
        id="c9"
      >
         
      </div>
    </div>
  </div>
</div>
`;

exports[`Die Component Snapshot testing 2 snapshot test 1`] = `
<div>
  <div
    class="pg-die"
  >
    <div
      class="row"
    >
      <div
        class="cell"
        id="c1"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c2"
      >
         
      </div>
      <div
        class="cell"
        id="c3"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c4"
      >
         
      </div>
      <div
        class="cell"
        id="c5"
      >
         
      </div>
      <div
        class="cell"
        id="c6"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c7"
      >
         
      </div>
      <div
        class="cell"
        id="c8"
      >
         
      </div>
      <div
        class="cell"
        id="c9"
      >
        🔴
      </div>
    </div>
  </div>
</div>
`;

exports[`Die Component Snapshot testing 3 snapshot test 1`] = `
<div>
  <div
    class="pg-die"
  >
    <div
      class="row"
    >
      <div
        class="cell"
        id="c1"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c2"
      >
         
      </div>
      <div
        class="cell"
        id="c3"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c4"
      >
         
      </div>
      <div
        class="cell"
        id="c5"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c6"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c7"
      >
         
      </div>
      <div
        class="cell"
        id="c8"
      >
         
      </div>
      <div
        class="cell"
        id="c9"
      >
        🔴
      </div>
    </div>
  </div>
</div>
`;

exports[`Die Component Snapshot testing 4 snapshot test 1`] = `
<div>
  <div
    class="pg-die"
  >
    <div
      class="row"
    >
      <div
        class="cell"
        id="c1"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c2"
      >
         
      </div>
      <div
        class="cell"
        id="c3"
      >
        🔴
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c4"
      >
         
      </div>
      <div
        class="cell"
        id="c5"
      >
         
      </div>
      <div
        class="cell"
        id="c6"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c7"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c8"
      >
         
      </div>
      <div
        class="cell"
        id="c9"
      >
        🔴
      </div>
    </div>
  </div>
</div>
`;

exports[`Die Component Snapshot testing 5 snapshot test 1`] = `
<div>
  <div
    class="pg-die"
  >
    <div
      class="row"
    >
      <div
        class="cell"
        id="c1"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c2"
      >
         
      </div>
      <div
        class="cell"
        id="c3"
      >
        🔴
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c4"
      >
         
      </div>
      <div
        class="cell"
        id="c5"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c6"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c7"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c8"
      >
         
      </div>
      <div
        class="cell"
        id="c9"
      >
        🔴
      </div>
    </div>
  </div>
</div>
`;

exports[`Die Component Snapshot testing 6 snapshot test 1`] = `
<div>
  <div
    class="pg-die"
  >
    <div
      class="row"
    >
      <div
        class="cell"
        id="c1"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c2"
      >
         
      </div>
      <div
        class="cell"
        id="c3"
      >
        🔴
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c4"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c5"
      >
         
      </div>
      <div
        class="cell"
        id="c6"
      >
        🔴
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c7"
      >
        🔴
      </div>
      <div
        class="cell"
        id="c8"
      >
         
      </div>
      <div
        class="cell"
        id="c9"
      >
        🔴
      </div>
    </div>
  </div>
</div>
`;

exports[`Die Component Snapshot testing Default snapshot 1`] = `
<div>
  <div
    class="pg-die"
  >
    <div
      class="row"
    >
      <div
        class="cell"
        id="c1"
      >
         
      </div>
      <div
        class="cell"
        id="c2"
      >
         
      </div>
      <div
        class="cell"
        id="c3"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c4"
      >
         
      </div>
      <div
        class="cell"
        id="c5"
      >
        🐽
      </div>
      <div
        class="cell"
        id="c6"
      >
         
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell"
        id="c7"
      >
         
      </div>
      <div
        class="cell"
        id="c8"
      >
         
      </div>
      <div
        class="cell"
        id="c9"
      >
         
      </div>
    </div>
  </div>
</div>
`;
